hummingbot/connector/exchange/btc_markets/btc_markets_exchange.py

```diff
@@ -0,0 +1,966 @@
+        self._throttler = AsyncThrottler(Constants.RATE_LIMITS)
+
+    @property
+    def name(self) -> str:
+        return "btc_markets"
+
+    @property
+    def order_books(self) -> Dict[str, OrderBook]:
+        return self._order_book_tracker.order_books
+
+    @property
+    def trading_rules(self) -> Dict[str, TradingRule]:
+        return self._trading_rules
+
+    @property
+    def in_flight_orders(self) -> Dict[str, BtcMarketsInFlightOrder]:
+        return self._in_flight_orders
+
+    @property
+    def status_dict(self) -> Dict[str, bool]:
+        """
+        A dictionary of statuses of various connector's components.
+        """
+        return {
+            "order_books_initialized": self._order_book_tracker.ready,
+            "account_balance": len(self._account_balances) > 0 if self._trading_required else True,
+            "trading_rule_initialized": len(self._trading_rules) > 0,
+            "user_stream_initialized":
+                self._user_stream_tracker.data_source.last_recv_time > 0 if self._trading_required else True,
+        }
+
+    @property
+    def ready(self) -> bool:
+        """
+        :return True when all statuses pass, this might take 5-10 seconds for all the connector's components and
+        services to be ready.
+        """
+        return all(self.status_dict.values())
+
+    @property
+    def limit_orders(self) -> List[LimitOrder]:
+        return [
+            in_flight_order.to_limit_order()
+            for in_flight_order in self._in_flight_orders.values()
+        ]
+
+    @property
+    def tracking_states(self) -> Dict[str, any]:
+        """
+        :return active in-flight orders in json format, is used to save in sqlite db.
+        """
+        return {
+            key: value.to_json()
+            for key, value in self._in_flight_orders.items()
+            if not value.is_done
+        }
+
+    def restore_tracking_states(self, saved_states: Dict[str, any]):
+        """
+        Restore in-flight orders from saved tracking states, this is st the connector can pick up on where it left off
+        when it disconnects.
+        :param saved_states: The saved tracking_states.
+        """
+        self._in_flight_orders.update({
+            key: BtcMarketsInFlightOrder.from_json(value)
+            for key, value in saved_states.items()
+        })
+
+    def supported_order_types(self) -> List[OrderType]:
+        """
+        :return a list of OrderType supported by this connector.
+        Note that Market order type is no longer required and will not be used.
+        """
+        return [OrderType.LIMIT, OrderType.LIMIT_MAKER]
+
+    def start(self, clock: Clock, timestamp: float):
+        """
+        This function is called automatically by the clock.
+        """
+        super().start(clock, timestamp)
+
+    def stop(self, clock: Clock):
+        """
+        This function is called automatically by the clock.
+        """
+        super().stop(clock)
```

Comment:
The start and stop methods are defined just to call super. That does not make sense. If they will have no especial logic please remove them.

Reply:
Based on crypto.com connector - https://github.com/hummingbot/hummingbot/blob/master/hummingbot/connector/exchange/crypto_com/crypto_com_exchange.py#L171